notification/events.go

```diff
@@ -81,6 +83,45 @@ type notificationHandler struct {
 	Ring *events.EventRing
 }
 
+// Check if notification can be send in the interval based on group by, returns true if it can sent
+func checkRepeatInterval(ctx context.Context, n NotificationWithSpec, event models.Event) (bool, error) {
+	validKeys := map[string]string{
+		"notification_id": n.ID.String(),
+		"resource_id":     event.Properties["id"],
+		"source_event":    event.Name,
+	}
+	var clauses []clause.Expression
+	for _, g := range n.GroupBy {
+		if val, exists := validKeys[g]; exists {
+			clauses = append(clauses, clause.Eq{Column: g, Value: val})
+		}
+	}
+	var lastSent time.Time
+	if len(clauses) > 0 {
+		tx := ctx.DB().Model(&models.NotificationSendHistory{}).Clauses(clauses...).Select("created_at").Scan(&lastSent)
+		if tx.Error != nil {
+			return false, fmt.Errorf("error querying db for last send notification[%s]: %w", n.ID, tx.Error)
+		}
+		// Allow notification to be sent as there is no history
+		if tx.RowsAffected == 0 {
+			return true, nil
+		}
+		if lastSent.IsZero() {
+			return false, fmt.Errorf("last sent not found for notification[%s]", n.ID)
+		}
+
+		interval, err := text.ParseDuration(n.RepeatInterval)
+		if err != nil {
+			return false, fmt.Errorf("error parsing repeat interval[%s] to time.Duration: %w", n.RepeatInterval, err)
+		}
+		if time.Since(lastSent) <= *interval {
```

Comment:
We are comparing db time and local time here - We should move this to the query

Reply:
Updated